### DISCUSSION TITLE
NewLint:  tx:=gorm.DB.Begin check tx.Error before use tx.

### DISCUSSION THREAD

**Discussion by jingyugao** [golangci/golangci-lint#1905] [Q&A]
I want to add a new lint to check this code.
```
tx:=db.Begin()
if tx.Error!=nil{ // if network failure, tx.Error will not nil. if not check this, we will get a panic in later code.
return tx.Error
}
// do something

```

**Comment by andig** [comment]
Same thing here. Just had this error in my code which panics on nil:

	if err.Error() == "408 Request Timeout" {
		err = nil
	}

I was wondering if there is an existing linter for this purpose?
